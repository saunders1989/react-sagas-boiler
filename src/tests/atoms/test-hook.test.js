import React from 'react';
import { shallow, mount } from 'enzyme';

import TestHook from '../../components/atoms/test-hook';

/** @test {TestHook} */
describe('TestHook', () => {

  it('should be defined', () => {
    expect(TestHook).toBeDefined();
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <TestHook />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when button is clicked', () => {
    const wrapper = mount(<TestHook />);
    const button = wrapper.find('button');
    button.simulate('click');

    expect(button.text()).toBe('close');
  });
});
