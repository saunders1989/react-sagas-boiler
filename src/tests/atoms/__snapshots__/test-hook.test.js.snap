// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestHook should render correctly 1`] = `
<div
  className="test-hook"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Open
  </button>
  <p />
  <div
    className="test-hook__draw"
  >
    <p>
      this is the draw
    </p>
    <p>
      this is the draw
    </p>
    <p>
      this is the draw
    </p>
  </div>
</div>
`;
